packages/toolkit/src/query/tests/buildHooks.test.tsx

```diff
@@ -2353,14 +2366,25 @@ describe('skip behaviour', () => {
     )
 
     expect(result.current).toEqual(uninitialized)
+    await delay(1)
     expect(subscriptionCount('getUser(1)')).toBe(0)
 
-    rerender([1])
-    expect(result.current).toMatchObject({ status: QueryStatus.pending })
+    await act(async () => {
+      rerender([1])
+    })
+    expect(result.current).toMatchObject({ status: QueryStatus.fulfilled })
+    await delay(1)
     expect(subscriptionCount('getUser(1)')).toBe(1)
 
-    rerender([1, { skip: true }])
-    expect(result.current).toEqual(uninitialized)
+    await act(async () => {
+      rerender([1, { skip: true }])
+    })
+    expect(result.current).toEqual({
+      ...uninitialized,
+      currentData: undefined,
```

Comment:
This is unexpected tbh. I'm pretty sure the hook should reset on changing to `skip`, as per the same logic that happens on a store reset. Something here is broken.
See
https://github.com/reduxjs/redux-toolkit/blob/dbcc556a2cd1ad6fd9776bc62131c9c43c72c4c0/packages/toolkit/src/query/react/buildHooks.ts#L608-L627